------------------------------------------------------------------------

---
title: "Data_Processing_TPP"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{Data_Processing_TPP}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```

```{r, include=TRUE}
devtools::load_all(".")

```

Library setup

```{r setup}

library(MSstats)
library(MSstatsConvert)
library(MSstatsTMT)
library(tidyr)
library(stringr)
library(TPP)
library(NPARC)
library(ggplot2)
library(ggrepel)
theme_set(theme_bw())

```

# Data Import

### Step 1: read_cetsa

```{r}
WD<-"~/work/ivanovlab/figueroa-navedo.a/OnePot_dataset/Reprocessed/"
setwd(WD)
PD_df<-read_cetsa(WD,WD,Frac=FALSE,temps=set_temps(11,c("37.3","40.2","44.9","47.6","51","54","57","61.8","64.1","67","75")),solvent="DMSO|V",CARRIER=FALSE)
```
Step 2 generate annotation file

```{r}

# #set mapping between temperatures and channels
temps<-set_temps(11,c("37.3","40.2","44.9","47.6","51","54","57","61.8","64.1","67","75"))
#TMT11 was used but the 75C temperature was filtered out
# temps$Channel<-ifelse(temps$Channel=="131","131C",temps$Channel)
#add temperatures
df_annotation<-PD_df|>dplyr::select(Spectrum.File,Channel,File.ID)|>dplyr::distinct()|>dplyr::inner_join(temps,by="Channel")
#Add bioreplicate
df_annotation$BioReplicate<-ifelse(df_annotation$temperature==min(df_annotation$temperature),"Norm",paste0(df_annotation$Channel,"_",df_annotation$treatment))

df_annotation$Mixture<-paste0(df_annotation$treatment,"_",
                                           df_annotation$TechRepMixture)


df_annotation$Condition<-ifelse(df_annotation$temperature==min(df_annotation$temperature),"Norm",paste0(df_annotation$Channel,"_",df_annotation$treatment))


df_annotation$Run<-df_annotation$Spectrum.File

#Filter out shared accessions
#df<-df[!stringr::str_detect(df$`Master Protein Accessions`,";"),]
#Filter out channel 131N
PD_df<-PD_df[!stringr::str_detect(PD_df$Channel,"131C"),]
#replace 131C with 131
PD_df$Channel<-ifelse(PD_df$Channel=="131N","131",PD_df$Channel)

PD_df$treatment<-stringr::str_extract(PD_df$Condition,"[[:lower:]]+")
PD_df<-PD_df|>dplyr::group_by(Accession,treatment,Channel)|>dplyr::mutate(TechRepMixture=seq(1,dplyr::n()))|>dplyr::ungroup()


```


Check missing value distributions

```{r}
mapping <-PD_df |>dplyr::select(Condition,Subject)|>
  dplyr::distinct()|>
  dplyr::group_by(Condition)|>
  dplyr::arrange(Condition)|>
  dplyr::mutate(TechRepMixture=seq(1,dplyr::n()),
                Color=ifelse(Condition=="vehicle"&TechRepMixture==1,"#00BCF4",
                             ifelse(Condition == "vehicle" & TechRepMixture==2,"#2b8cbe",
                                    ifelse(Condition=="treated" & TechRepMixture==1,"#F8776D",
                                            "#fa9fb5"))))|>
  dplyr::ungroup()

MV<-PD_df|>
  dplyr::inner_join(mapping)|>
  dplyr::filter(TechRepMixture==1|TechRepMixture==2)|>
  dplyr::group_by(Accession,Subject)|>
  dplyr::select(Accession,Subject,Channel,temperature,Abundance,Condition,TechRepMixture,Color)|>
  dplyr::distinct()|>
  dplyr::filter(Channel!="131C")|>
  dplyr::mutate(missing_channels=sum(is.na(Abundance)),
                plex = paste0(Condition,"_",TechRepMixture))|>
  dplyr::filter(missing_channels!=10)|>
  dplyr::select(Accession,missing_channels,plex,Condition,Color)|>
  dplyr::distinct()
  

histogram_mv<-ggplot(MV,mapping=aes(x=missing_channels,fill=plex),group=Condition)+geom_histogram()+facet_wrap(~Condition)+xlab("# of missing channels")+xlim(0,10)+
  scale_x_continuous(breaks=seq(0,10,1),labels=seq(0,10,1),limits= c(-1,10))+
  ylab("# of Proteins")+
  scale_fill_manual(values = mapping$Color)+theme(text=element_text(size =20))
  
path = getwd()
histogram_mv
ggplot2::ggsave(path = path, width = 24, height = 6.5, device='tiff', dpi=600, filename = "MV_Dataset3a.tiff")


```
# Calculate proteins with more than 3 missing values
```{r}
PD_df_filtered<-PD_df |>
  dplyr::inner_join(MV|>
                      dplyr::select(Accession,Subject,plex,missing_channels)|>
                      dplyr::rename(Experiment=plex)|>
                      dplyr::distinct())|>
  dplyr::filter(missing_channels >=3)
proteins_missing<-length(unique(PD_df_filtered$Accession))
print(paste0(round(proteins_missing/length(unique(PD_df$Accession))*100,2),"% of proteins were filtered out with more than 3 missing values"))
```
# Data processing with TPP

1.  Normalize the data

```{r}
#remove shared proteins and isoforms
PD_df<-PD_df[!stringr::str_detect(PD_df$Accession,";|-"),]

Human_TPPnorm_Proteins<-TPP_normalization(PD_df,TPPfilters=TRUE,temps=temps,reference="126",CARRIER=FALSE)

df<-list(ProteinLevelData=Human_TPPnorm_Proteins$normData|>as.data.frame())

no_norm<-df$ProteinLevelData
no_norm<-no_norm[!no_norm$Condition=="Norm",]
no_norm<-list(ProteinLevelData=no_norm)
```

3.  Create a contrast matrix to measure the average treatment effect

    ```{r}
    df3<-list()
    #remove reference channel because there is no variability after protein normalization
    df3$ProteinLevelData<-df$ProteinLevelData|>
      as.data.frame()|>
      dplyr::mutate(BioReplicate=as.character(BioReplicate),
                    temperature=as.character(temperature),
                    TechRepMixture=as.numeric(TechRepMixture),
        treatment=stringr::str_to_lower(Experiment,"[[:lower:]]+"))


    #choose three ranges around the midpoint
        temps_late<-c("54","57","61.8")
        temps_mid<-c("47.6","51","54")
        temps_ear<-c("40.2","47.6","51")
        #then create a contrast matrix without the reference channel 
        comparison_late<-make_contrast_matrix(df3,temps=temps_late)
        comparison_mid<-make_contrast_matrix(df3,temps=temps_mid)
        comparison_ear<-make_contrast_matrix(df3,temps=temps_ear)
      df3$ProteinLevelData$temperature<-as.numeric(df3$ProteinLevelData$temperature)
    ```

4.  Fit linear mixed models with groupComparisons from MSstatsTMT

```{r}
DIM_late<-groupComparisonThermalProfiling(
  df3,
  contrast.matrix = comparison_late,
  moderated = FALSE,
  adj.method = "BH",
  remove_norm_channel = TRUE,
  remove_empty_channel = TRUE,
  save_fitted_models = TRUE,
  use_log_file = FALSE,
  append = FALSE,
  verbose = TRUE,
  log_file_path = NULL,
  missing_timepoint = "replace",
  replacement=list("131_vehicle" = c("130C_vehicle"))
)

DIM_mid<-groupComparisonThermalProfiling(
  df3,
  contrast.matrix = comparison_mid,
  moderated = FALSE,
  adj.method = "BH",
  remove_norm_channel = TRUE,
  remove_empty_channel = TRUE,
  save_fitted_models = TRUE,
  use_log_file = FALSE,
  append = FALSE,
  verbose = TRUE,
  log_file_path = NULL,
   missing_timepoint = "replace",
  replacement=list("131_vehicle" = c("130C_vehicle"))
)

DIM_ear<-groupComparisonThermalProfiling(
  df3,
  contrast.matrix = comparison_ear,
  moderated = FALSE,
  adj.method = "BH",
  remove_norm_channel = TRUE,
  remove_empty_channel = TRUE,
  save_fitted_models = TRUE,
  use_log_file = FALSE,
  append = FALSE,
  verbose = TRUE,
  log_file_path = NULL,
   missing_timepoint = "replace",
  replacement=list("131_vehicle" = c("130C_vehicle"))
)
```

# Fit models using filtered data which only considers the variance from the selected contrasts

```{r}
early_data<-list()
middle_data<-list()
later_data<-list()
comparison_early<-comparison_ear[which(!comparison_ear==0)]|>as.matrix()|>t()
colnames(comparison_early)<-colnames(comparison_ear)[which(!comparison_ear==0)]
rownames(comparison_early)<-"DIM"
early_data$ProteinLevelData<-df3$ProteinLevelData[df3$ProteinLevelData$Condition%in%colnames(comparison_early),]
comparison_middle<-comparison_mid[which(!comparison_mid==0)]|>as.matrix()|>t()
rownames(comparison_middle)<-"DIM"
colnames(comparison_middle)<-colnames(comparison_mid)[which(!comparison_mid==0)]
middle_data$ProteinLevelData<-df3$ProteinLevelData[df3$ProteinLevelData$Condition%in%colnames(comparison_middle),]
comparison_later<-comparison_late[which(!comparison_late==0)]|>as.matrix()|>t()
rownames(comparison_later)<-"DIM"
colnames(comparison_later)<-colnames(comparison_late)[which(!comparison_late==0)]
later_data$ProteinLevelData<-df3$ProteinLevelData[df3$ProteinLevelData$Condition%in%colnames(comparison_later),]

DIM_later<-MSstatsTMT::groupComparisonTMT(
later_data,
contrast.matrix = comparison_later,
moderated = FALSE,
adj.method = "BH",
remove_norm_channel = TRUE,
remove_empty_channel = TRUE,
save_fitted_models = TRUE,
use_log_file = FALSE,
append = FALSE,
verbose = TRUE,
log_file_path = NULL
)
DIM_middle<-MSstatsTMT::groupComparisonTMT(
middle_data,
contrast.matrix = comparison_middle,
moderated = FALSE,
adj.method = "BH",
remove_norm_channel = TRUE,
remove_empty_channel = TRUE,
save_fitted_models = TRUE,
use_log_file = FALSE,
append = FALSE,
verbose = TRUE,
log_file_path = NULL
)
DIM_early<-MSstatsTMT::groupComparisonTMT(
early_data,
contrast.matrix = comparison_early,
moderated = FALSE,
adj.method = "BH",
remove_norm_channel = TRUE,
remove_empty_channel = TRUE,
save_fitted_models = TRUE,
use_log_file = FALSE,
append = FALSE,
verbose = TRUE,
log_file_path = NULL
)
```

# Benchmarking

## TPP

We start by using the TPP method

```{r}
proteins1<-df$ProteinLevelData|>dplyr::mutate(Experiment=as.character(Experiment),treatment=stringr::str_extract(Experiment,"[[:lower:]]+"))
#QC profile to check the boxplot of data
# ggplot(proteins1,mapping=aes(x=as.factor(temperature),y=Abundance))+geom_boxplot()+ylim(0,1)

# proteins1<-filter_good_data(proteins)
#Run TPP spline and implement the NPARC F-test with 5 DF
start=proc.time()
TPPtr_human_results = TPP_NPARC_calc(proteins1,method="TPP",DF=5,CARRIER=FALSE,temps=set_temps(10,c(37.3, 40.6, 43.9, 47.2, 50.5, 53.8, 57.1, 60.4, 64, 67)),NORM=FALSE,filters=TRUE)
end=proc.time()
print(end-start)
```

## TPP splines

```{r}
start=proc.time()
TPP_human_results_F = TPP_NPARC_calc(proteins1,method="NPARC",DF=5,CARRIER=FALSE,temps=set_temps(10,c(37.3, 40.6, 43.9, 47.2, 50.5, 53.8, 57.1, 60.4, 64, 67)),NORM=FALSE,filters=TRUE)
end=proc.time()
print(end-start)
```

## NPARC

```{r}

#Then run NPARC
NPARC_human_fits <-NPARC::NPARCfit(x =proteins1$temperature,
                 y =proteins1$Abundance,
                 id =proteins1$Protein,
                 groupsNull = NULL,
                 groupsAlt = as.character(proteins1$treatment),
                 returnModels = TRUE)
```

## Empirical test

```{r}
NPARC_test_emp<-NPARC::NPARCtest(NPARC_human_fits$metrics,dfType="empirical")
hist(NPARC_test_emp$pVal)
```

## Theoretical test

```{r}
NPARC_test_theo<-NPARC::NPARCtest(NPARC_human_fits$metrics,dfType="theoretical")
hist(NPARC_test_theo$pVal)


```

## SCAM Splines

```{r}

SCAM_Human_proteins<-MSstatstoSCAM(proteins1)
head(SCAM_Human_proteins)

```

```{r}
DIM_Human_SCAM<-compute_pvalues_ATE_RE_F(SCAM_Human_proteins)

```

# Determine which accessions are kinase targets

```{r}
WD<-"~/work/ivanovlab/figueroa-navedo.a/OnePot_dataset/CETSA_onePot/OnePot/"
h<-list.files(WD,pattern="Proteins.xlsx")|>as.list()
setwd(WD)
Proteins<-data.frame()
Proteins<-lapply(h,function(x) readxl::read_xlsx(paste0(WD,x))) |> dplyr::bind_rows()|>dplyr::select(Accession,"Gene Symbol")|>dplyr::rename("GeneSymbol"="Gene Symbol")|>dplyr::distinct()
#Read in human kinome (536 proteins)
kinases<-read.csv(paste0(WD,"/HumanKinome_040920.csv"))|>dplyr::select(Accession)
kinases<-kinases$Accession


#remove isoforms from UniProt accessions
df<-df$ProteinLevelData|>dplyr::mutate(Accession=stringr::str_remove(uniqueID,"-[[:digit:]]+"))|>dplyr::distinct()


#join Uniprot IDs to original DF
kinase_df<-kinases
```

# Bar charts

For the total number of proteins

```{r}

Total_prot<-list("TPP"=unique(PD_df$Accession),"NPARC"=unique(PD_df$Accession),"SCAM"=unique(PD_df$Accession),"MSstatsTMT"=unique(PD_df$Accession))
#remove shared accessions
Total_prot<-lapply(Total_prot,function(x) x[!stringr::str_detect(x,";")])
#remove missing values
Total_prot<-lapply(Total_prot,function(x) as.character(x)|>na.omit())

#verify if targets are present in the Statistical model fit convergence
Benchmarks_TPPproc_targ<-purrr::map(Total_prot,function(x) sum(as.character(x)%in%kinases))

Total_prot_n<-purrr::map2(Total_prot,Benchmarks_TPPproc_targ,function(x,y) data.frame(Proteins=as.character(length(unique(x))),ver_targets=as.character(y)))|>dplyr::bind_rows()|>dplyr::mutate(Steps="Proteins (unique protein groups):\n from PD",Analysis=c("TPP","NPARC","SCAM","MSstatsTMT"))
```

output from TPP processing

```{r}
Output_proc<-list("TPP"=unique(Human_TPPnorm_Proteins$normData$Protein),"NPARC"=unique(Human_TPPnorm_Proteins$normData$Protein),"SCAM"=unique(Human_TPPnorm_Proteins$normData$Protein),"MSstatsTMT"=unique(Human_TPPnorm_Proteins$normData$Protein))
#remove shared accessions
Output_proc<-lapply(Output_proc,function(x) x[!stringr::str_detect(x,";")])
#remove missing values
Output_proc<-lapply(Output_proc,function(x) as.character(x)|>na.omit())

#verify if targets are present in the Statistical model fit convergence
Benchmarks_TPPproc_targ<-purrr::map(Output_proc,function(x) sum(x%in%kinases))

Output_proc_n<-purrr::map2(Output_proc,Benchmarks_TPPproc_targ,function(x,y) data.frame(Proteins=as.character(length(unique(x))),ver_targets=as.character(y)))|>dplyr::bind_rows()|>dplyr::mutate(Steps="Proteins: after \n TPP normalization (Table 1)",Analysis=c("TPP","NPARC","SCAM","MSstatsTMT"))
```

Statistical model fit convergence

```{r}
Benchmarks_TPPproc<-list("TPP"=unique(TPP_human_results_F$uniqueID[!is.na(TPP_human_results_F$p_adj_NPARC)]),"NPARC"=unique(as.character(NPARC_test_emp$id)[!is.na(NPARC_test_emp$pAdj)]),"SCAM"=unique(as.character(DIM_Human_SCAM$Accession[!is.na(DIM_Human_SCAM$ATE_padjBH)])),"MSstatsTMT"=unique(as.character(DIM_late$ComparisonResult$Protein[is.na(DIM_late$ComparisonResult$issue)])))
#remove shared accessions
Benchmarks_TPPproc<-lapply(Benchmarks_TPPproc,function(x) x[!stringr::str_detect(x,";")])
#remove missing values
Benchmarks_TPPproc<-lapply(Benchmarks_TPPproc,function(x) as.character(x)|>na.omit())

#Benchmarks_TPPproc<-purrr::map2(Benchmarks_TPPproc,Output_proc,function(x,y)x[x%in%y])
#verify if targets are present in the Statistical model fit convergence
Benchmarks_TPPproc_targ<-purrr::map(Benchmarks_TPPproc,function(x) sum(x%in%kinases))

Benchmarks_TPPproc_n<-purrr::map2(Benchmarks_TPPproc,Benchmarks_TPPproc_targ,function(x,y) data.frame(Proteins=as.character(length(unique(x))),ver_targets=as.character(y)))|>dplyr::bind_rows()|>dplyr::mutate(Steps="Proteins: converged fit\n(Table 2, steps 1-3)",Analysis=c("TPP","NPARC","SCAM","MSstatsTMT"))
```

For proteins that passed the R\^2 and plateau filters

```{r}
TPPtr_filters<-unique(TPPtr_human_results$Protein_ID[TPPtr_human_results$passed_filter_vehicle_1_vs_treated_1&TPPtr_human_results$passed_filter_vehicle_2_vs_treated_2])
#remove shared accessions
TPPtr_filters<-TPPtr_filters[!stringr::str_detect(TPPtr_filters,";")]
#remove missing values
TPPtr_filters<-TPPtr_filters|>na.omit()


Benchmarks_TPPproc_filt<-purrr::map2(Benchmarks_TPPproc,names(Benchmarks_TPPproc),function(x,y) ifelse(y=="NPARC",sum(x %in% TPPtr_filters),sum(x%in%x)))

Benchmarks_TPPproc_filt<-lapply(Benchmarks_TPPproc_filt,function(x) as.character(x)|>na.omit())
#verify if targets are present in the filtered data
Benchmarks_TPPproc_targ<-
  purrr::map2(Benchmarks_TPPproc,names(Benchmarks_TPPproc),function(x,y) ifelse(y=="NPARC",sum(x %in% TPPtr_filters& x%in%kinases),sum(x%in%kinases)))

Benchmarks_TPPproc_filt_n<-purrr::map2(Benchmarks_TPPproc_filt,Benchmarks_TPPproc_targ,function(x,y) data.frame(Proteins=as.character(x),ver_targets=as.character(y)))|>dplyr::bind_rows()|>dplyr::mutate(Steps="Proteins: after filtering \n(Table 2, steps 1-4)",Analysis=c("TPP","NPARC","SCAM","MSstatsTMT"))
```

For differentially expressed proteins

```{r}
Benchmarks_TPPproc_p<-list("TPP"=unique(TPP_human_results_F$uniqueID[TPP_human_results_F$p_adj_NPARC<0.05]),#R2 and plateau filters
                           "NPARC"=unique(as.character(NPARC_test_emp$id)[NPARC_test_emp$pAdj<0.05&as.character(NPARC_test_emp$id) %in% TPPtr_filters]),"SCAM"=unique(DIM_Human_SCAM$Accession[DIM_Human_SCAM$ATE_padjBH<0.05]),"MSstatsTMT"=unique(DIM_late$ComparisonResult$Protein[DIM_late$ComparisonResult$adj.pvalue<0.05]))
#remove shared accessions
Benchmarks_TPPproc_p<-lapply(Benchmarks_TPPproc_p,function(x) x[!stringr::str_detect(x,";")])
#remove missing values
Benchmarks_TPPproc_p<-lapply(Benchmarks_TPPproc_p,function(x) as.character(x)|>na.omit())

#verify if differentially abundant proteins are present in the filtered data
Benchmarks_TPPproc_targ<-
  purrr::map2(Benchmarks_TPPproc_p,names(Benchmarks_TPPproc_p),function(x,y) ifelse(y=="NPARC",sum(x %in% TPPtr_filters&x%in%kinases),sum(x%in%kinases)))

Benchmarks_TPPproc_p_n<-purrr::map2(Benchmarks_TPPproc_p,Benchmarks_TPPproc_targ,function(x,y) data.frame(Proteins=as.character(length(unique(x))),ver_targets=as.character(y)))|>dplyr::bind_rows()|>dplyr::mutate(Steps="Proteins: differential abundance \n(Table 2, steps 1-5)",Analysis=c("TPP","NPARC","SCAM","MSstatsTMT"))
```

## Plot the bar charts

```{r}
bar_human<-dplyr::bind_rows(Total_prot_n,Output_proc_n,Benchmarks_TPPproc_n,Benchmarks_TPPproc_filt_n,Benchmarks_TPPproc_p_n)

bar_human$Analysis<-factor(bar_human$Analysis,levels=c("TPP","NPARC","SCAM","MSstatsTMT"))

col<-data.frame(Analysis=factor(unique(bar_human$Analysis),levels=c("TPP","NPARC","SCAM","MSstatsTMT")),"stat_model"=c("#FEC44F","#D95F0E","#2C7FB8","#07fff8"))


bar_human<-bar_human|>dplyr::inner_join(col)

bar_human$Analysis<-factor(paste0("TPP proc., ",bar_human$Analysis," model"),levels=c("TPP proc., TPP model","TPP proc., NPARC model","TPP proc., SCAM model","TPP proc., MSstatsTMT model"))


bar_human$Steps<-factor(bar_human$Steps,levels=c("Proteins: differential abundance \n(Table 2, steps 1-5)","Proteins: after filtering \n(Table 2, steps 1-4)","Proteins: converged fit\n(Table 2, steps 1-3)","Proteins: after \n TPP normalization (Table 1)","Proteins (unique protein groups):\n from PD"))


bar_human$Protein_targ<-paste0(bar_human$Proteins," ","(",bar_human$ver_targets,")")

bHuman<-ggplot(bar_human|>dplyr::group_by(Analysis),mapping=aes(x=Steps,y=as.integer(Proteins),alpha=Steps,fill=stat_model))+geom_col(stat_count=aes(x=Steps,y=Proteins),colour="black")+scale_y_continuous(limits=c(0,15000),breaks = c(0,6000,12000))+ylab("Number of proteins remaining after each step")+coord_flip()+facet_wrap(~Analysis,nrow=1, labeller = label_wrap_gen(width=15))+scale_fill_manual(values=c("#2C7FB8","#07fff8","#FEC44F","#D95F0E"))+
  xlab("Steps")+
  geom_text(aes(label=Protein_targ),size=8,position=position_dodge(width=0.5),hjust=-0.150,alpha=0.9)+theme(text=element_text(size=25,                          color = "black"),axis.text.x = element_text(size=25,face="bold"),axis.text.y=element_text(size=25,face="bold"),
                                                                                                            axis.title.x = element_text(size=25),axis.title.y=element_text(size=25),strip.text.x=element_text(size=25),legend.position="none", strip.placement = "outside")

bHuman
ggplot2::ggsave(path = path, width = 24, height = 6.5, device='tiff', dpi=600, filename = "Bar_charts_TPP.tiff")
```

Save the proteins processed by MSstatsTMT

```{r}
Benchmarks_TPPproc_p<-lapply(Benchmarks_TPPproc_p,function(x) as.character(x)|>na.omit())

names(Benchmarks_TPPproc_p)<-c("TPP proc.
TPP stat. model","TPP proc. 
NPARC stat. model","TPP proc.
SCAM stat. model","TPP proc.
MSstatsTMT stat. model")
Thermallate_spline_TPPprocesed<-Benchmarks_TPPproc_p
saveRDS(Thermallate_spline_TPPprocesed,"Thermallate_spline_TPPprocesed.RDS")
```

# Venn Diagrams

```{r}

Venn_Humandata<-ggvenn::ggvenn(data=Benchmarks_TPPproc_p,fill_color =c("#D95F0E","#FEC44F","#07fff8", "#2C7FB8"),set_name_size=4,stroke_alpha=2,text_size=4,fill_alpha = 0.6,show_percentage = FALSE)+ggtitle("pAdj < 0.05")+theme(text=element_text(size=20))+coord_cartesian(ylim = c(-2,2),xlim = c(-4,4))
Venn_Humandata
```

# Venn Diagrams (top 3)

```{r}
png(filename = "Venn_TPPcurve_spline_TPPproc.png",
    width =12, height =6, units = "in", pointsize = 12,
    res = 600)  

library(ggvenn)
ggvenn::ggvenn(data=Benchmarks_TPPproc_p[c(1,2,4)],fill_color =c("#D95F0E","#FEC44F","#2C7FB8"),set_name_size=5,text_size=8,stroke_size=1.5,fill_alpha = 0.6,show_percentage = FALSE)+ggtitle("TPP processed pAdj < 0.05")+theme(text=element_text(size=20))+coord_cartesian(ylim = c(-2,2),xlim = c(-4,4))

dev.off() 

```


Figure 8

```{r}

Benchmarks_external<-list("3a: TPPproc. 
MSstatsTMT stat. model"=Benchmarks_TPPproc_p$`MSstatsTMT stat. model`,
                          "Reference set"=na.omit(kinases),
                          "3a: MSstatsTMT proc. 
MSstatsTMT stat. model"=Thermal_MSstatsTMTprocesed$`MSstatsTMT stat. model`)

png(filename = "Venn_thermal_MSstatsTMTstatmodel_processingdiff_kinases.png",
    width =600, height = 600, units = "px", pointsize = 12,
    res = 130,type ="cairo")  

library(ggvenn)
ggvenn::ggvenn(data=Benchmarks_external,fill_color =c("#2C7FB8","#7b057b","#2C7FB8"),set_name_size=5,text_size=8,stroke_size=1.5,fill_alpha = 0.6,show_percentage = FALSE)+ggtitle("kinases pAdj < 0.05")+theme(text=element_text(size=20))

dev.off() 
```

Filter expected targets from each result

```{r}
NPARC_human<-NPARC_test_emp|>dplyr::filter(id %in% kinases)|>dplyr::mutate(Protein=id)

TPP_human<-TPP_human_results_F|>dplyr::filter(uniqueID %in% kinases)|>dplyr::mutate(Protein=uniqueID)

SCAM_human<-DIM_Human_SCAM|>dplyr::filter(Accession %in% kinases)|>dplyr::mutate(Protein=Accession)

MSstat_human<-DIM_late$ComparisonResult|>dplyr::filter(Protein%in%kinases)

Stat_mod<-dplyr::bind_rows(proteins1)|>as.data.frame()|>dplyr::filter(Protein%in% c("P36507","Q02750","Q13188","O14757"))|>dplyr::mutate(Accession=Protein)|>dplyr::left_join(TPP_human)|>dplyr::left_join(NPARC_human|>dplyr::select(-df1,-df2))|>dplyr::left_join(SCAM_human)|>dplyr::left_join(MSstat_human)|>dplyr::group_by(Accession)|>dplyr::mutate(NPARC_adjpval=pAdj,TPP_adjpval=p_adj_NPARC,SCAM_adjpval=ATE_padjBH,MSstats_adjpval=adj.pvalue)|>dplyr::group_split()


```

Append MSstatsTMT fitted models to the data

```{r}

DIM_targets<-DIM_late$FittedModel[which(names(DIM_late$FittedModel) %in% c("P36507","Q02750","Q13188","O14757"))]
fitted_MSstat<-purrr::map2(Stat_mod,DIM_targets,function(x,y)x|>dplyr::mutate(model=list(y)))|>dplyr::bind_rows()
```

# Plot the curves

```{r}
#identify significant kinases per method
targets<-fitted_MSstat|>dplyr::mutate(treatment=stringr::str_extract(Experiment,"[[:lower:]]+"))|>dplyr::group_by(Protein)|>dplyr::group_split()

nam<-as.character(unique(dplyr::bind_rows(targets)$Protein))
target_curves<-lapply(targets,function(x){plotMethods(x,temps=c("54","57","61.8"),labels=c("A","B","C","D"),processing="TPP",fit="Spline")
})
names(target_curves)<-nam

#set path
path<-getwd()

target_curves$P36507
ggplot2::ggsave(path = path, width = 24, height = 6.5, device='tiff', dpi=600, filename = "Curves_TPP_Spline_P36507.tiff")

target_curves$Q02750
ggplot2::ggsave(path = path, width = 24, height = 6.5, device='tiff', dpi=600, filename = "Curves_TPP_Spline_Q02750.tiff")

target_curves$Q13188
ggplot2::ggsave(path = path, width = 24, height = 6.5, device='tiff', dpi=600, filename = "Curves_TPP_Spline_Q13188.tiff")

target_curves$O14757
ggplot2::ggsave(path = path, width = 24, height = 6.5, device='tiff', dpi=600, filename = "Curves_TPP_Spline_O14757.tiff")
```

## Compare Venn Diagrams between processing

```{r}

MSstatsTMT_proc<-lapply(Thermal_mid_MSstatsTMTprocesed,
                        function(x) as.character(x)|>na.omit())

TPP_proc<-lapply(Benchmarks_TPPproc_p,function(x) as.character(x)|>na.omit())


compare_proc<-purrr::map2(TPP_proc,MSstatsTMT_proc,function(x,y) list("MSstatsTMT proc."=y,"TPP proc."=x))


colors<-list(TPP=c("#D95F0E","#D95F0E"),NPARC=c("#FEC44F","#FEC44F"),SCAM=c("#07fff8","#07fff8"),MSstatsTMT=c("#2C7FB8","#2C7FB8"))

colors<-purrr::set_names(colors,names(colors))
purrr::map2(compare_proc[1],colors[1],function(x,y)VennDiagram::venn.diagram(
  x = x,
  category.names =names(x),
  filename = paste0('Venn_Human_processing ',paste("TPP")),
  output = TRUE ,
  imagetype="png" ,
  height = 528 , 
  width = 528 , 
  resolution = 600,
  compression = "lzw",
  lwd = 1,
  col=y,
   cex=0.5,
  cat.cex=0.3,
  rotation.degree=-90,
     cat.pos = c(180,0),
          cat.dist = c(0.02, 0.02),
  scaled=FALSE,
  inverted=FALSE
  
))
  

purrr::map2(compare_proc[2],colors[2],function(x,y)VennDiagram::venn.diagram(
  x = x,
  category.names =names(x),
  filename = paste0('Venn_Human_processing ',paste("NPARC")),
  output = TRUE ,
  imagetype="png" ,
  height = 528 , 
  width = 528 , 
  resolution = 600,
  compression = "lzw",
  lwd = 1,
  col=y,
  cex=0.5,
  cat.cex=0.3,
  rotation.degree=-90,
    cat.pos = c(180,0),
    cat.dist = c(0.02, 0.02),
  scaled=FALSE,
  inverted=FALSE
  
))


purrr::map2(compare_proc[3],colors[3],function(x,y)VennDiagram::venn.diagram(
  x = x,
  category.names =names(x),
  filename = paste0('Venn_Human_processing ',paste("SCAM")),
  output = TRUE ,
  imagetype="png" ,
  height = 528 , 
  width = 528 , 
  resolution = 600,
  compression = "lzw",
  lwd = 1,
  col=y,
  cex=0.5,
  cat.cex=0.3,
  rotation.degree=90,
    cat.pos = c(180,0),
    cat.dist = c(0.02, 0.02),
  scaled=FALSE,
  inverted=FALSE
  
))

purrr::map2(compare_proc[4],colors[4],function(x,y)VennDiagram::venn.diagram(
  x = x,
  category.names =names(x),
  filename = paste0('Venn_Human_processing ',paste("MSstatsTMT")),
  output = TRUE ,
  imagetype="png" ,
  height = 528 , 
  width = 528 , 
  resolution = 600,
  compression = "lzw",
  lwd = 1,
  col=y,
  cex=0.5,
  cat.cex=0.3,
  rotation.degree=-90,
     cat.pos = c(180,0),
          cat.dist = c(0.02, 0.02),
  scaled=FALSE,
  inverted=FALSE
  
))
```

# Generate differential abundance table using three configurations for three temperatures

```{r}

results<-data.frame(processing=rep("MSstatsTMT",3),"stat. model"=rep("MSstatsTMT",3),configuration=c("late","mid","early"),temperatures=stringr::str_c(temps_ear,temps_mid,temps_late,sep=","),"p-adj lt 0.05"=c(length(na.omit(unique(DIM_late$ComparisonResult$Protein[DIM_late$ComparisonResult$adj.pvalue<0.05]))),
                                                                                                                                                            length(unique(na.omit(DIM_mid$ComparisonResult$Protein[DIM_mid$ComparisonResult$adj.pvalue<0.05]))),
                                                                                                                                                            length(unique(na.omit(DIM_ear$ComparisonResult$Protein[DIM_ear$ComparisonResult$adj.pvalue<0.05])))),
                    targets=c(sum(DIM_late$ComparisonResult$adj.pvalue<0.05&DIM_late$ComparisonResult$Protein%in%kinases),
                              sum(DIM_mid$ComparisonResult$adj.pvalue<0.05&DIM_mid$ComparisonResult$Protein%in%kinases),sum(DIM_ear$ComparisonResult$adj.pvalue<0.05&DIM_ear$ComparisonResult$Protein%in%kinases)))
                                                                                                         results     
```

## Now generate the same results for the filtered data with only 3 temperatures

```{r}
results_filtered<-data.frame(processing=rep("MSstatsTMT",3),"stat. model"=rep("MSstatsTMT",3),configuration=c("late","mid","early"),temperatures=stringr::str_c(temps_ear,temps_mid,temps_late,sep=","),"p-adj lt 0.05"=c(length(na.omit(unique(DIM_later$ComparisonResult$Protein[DIM_later$ComparisonResult$adj.pvalue<0.05]))),
                                                                                                                                                            length(unique(na.omit(DIM_middle$ComparisonResult$Protein[DIM_middle$ComparisonResult$adj.pvalue<0.05]))),
                                                                                                                                                            length(unique(na.omit(DIM_early$ComparisonResult$Protein[DIM_early$ComparisonResult$adj.pvalue<0.05])))),
                    targets=c(sum(DIM_later$ComparisonResult$adj.pvalue<0.05&DIM_later$ComparisonResult$Protein%in%kinases),
                              sum(DIM_middle$ComparisonResult$adj.pvalue<0.05&DIM_middle$ComparisonResult$Protein%in%kinases),sum(DIM_early$ComparisonResult$adj.pvalue<0.05&DIM_early$ComparisonResult$Protein%in%kinases)))
                                                                                                         results_filtered     
```

# Volcano plots for MSstatsTMT

```{r}
f<-DIM_late$ComparisonResult
f$diffexpressed <- "Not Shifted"
    f$diffexpressed[f$adj.pvalue < 0.05] <- "Shifted adj. pVal < 0.05"
    f$diffexpressed[f$adj.pvalue < 0.01] <- "Shifted adj. pVal < 0.01"
    f$diffexpressed<-as.factor(f$diffexpressed)
    f$targets<-NA
    f$delabel <- NA
    f$others<-NA
    
    f$others[which(f$adj.pvalue<0.01)]<-as.character(f$Protein[which(f$adj.pvalue<0.01)])
    f$targets <-ifelse(f$Protein %in%"Q13188-2","Q13188-2",ifelse(f$Protein %in% kinases,kinases,NA))
    f$delabel[f$Protein %in%kinases] <- as.character(f$Protein[f$Protein %in%kinases])
    flevels<-data.frame(colors=c("#762a83","#b35806","#1b7837"),
                        labels=c("Shifted adj. pVal < 0.05","Not Shifted","Shifted adj. pVal < 0.01"))
    flevels<-flevels %>% dplyr::filter(labels %in% unique(f$diffexpressed))

ggplot(f,mapping=aes(x=log2FC,y=log10(adj.pvalue),color=diffexpressed))+geom_point()+xlim(-2,2)+xlab("DIM")+scale_y_reverse()+geom_label_repel(f,mapping=aes(label=targets),color="black",max.overlaps = getOption("ggrepel.max.overlaps", default = 10),
nudge_x = 3,
nudge_y=1,
force = 3)+ylab("log(adj.pvalue)")+xlab("DIM")+theme(text=element_text(size=18))

```

```{r}
data.frame(Method="MsstatsTMT","Not.Shifted"=sum(as.numeric(f$diffexpressed=="Not Shifted")),"Shifted"=sum(as.numeric(!f$diffexpressed=="Not Shifted")))
```

# Volcano plots for SCAM

```{r}


f<-DIM_Human_SCAM
f$diffexpressed <- "Not Shifted"
    f$diffexpressed[f$ATE_padjBH < 0.05] <- "Shifted adj. pVal < 0.05"
    f$diffexpressed[f$ATE_padjBH < 0.01] <- "Shifted adj. pVal < 0.01"
    f$diffexpressed<-as.factor(f$diffexpressed)
    f$targets<-NA
    f$delabel <- NA
    f$others<-NA
    
    # f$others[which(f$ATE_padjBH<0.01)]<-as.character(f$Protein[which(f$ATE_padjBH<0.01)])
    f$targets <-ifelse(f$Accession %in%"Q13188-2","Q13188-2",ifelse(f$Accession %in% kinases,kinases,NA))
    f$delabel[f$Accession %in%kinases] <- as.character(f$Accession[f$Accession %in%kinases])
    flevels<-data.frame(colors=c("#762a83","#b35806","#1b7837"),
                        labels=c("Shifted adj. pVal < 0.05","Not Shifted","Shifted adj. pVal < 0.01"))
    flevels<-flevels %>% dplyr::filter(labels %in% unique(f$diffexpressed))

ggplot(f,mapping=aes(x=estimate,y=log10(ATE_padjBH),color=diffexpressed))+geom_point()+xlim(-2,2)+xlab("DIM")+scale_y_reverse()+geom_label_repel(f,mapping=aes(label=targets),color="black",max.overlaps = getOption("ggrepel.max.overlaps", default = 10),
nudge_x = 3,
nudge_y=1,
force = 3)+ylab("log(adj.pvalue)")+xlab("DIM SCAM")+theme(text=element_text(size=18))
```

Zoom in

```{r}
ggplot(f,mapping=aes(x=estimate,y=log10(ATE_padjBH),color=diffexpressed))+geom_point()+xlim(-2,2)+xlab("DIM")+scale_y_reverse()+geom_label_repel(f,mapping=aes(label=targets),color="black",max.overlaps = getOption("ggrepel.max.overlaps", default = 10),
nudge_x = 3,
nudge_y=1,
force = 3)+ylab("log(adj.pvalue)")+xlab("DIM")+theme(text=element_text(size=18))+ylim(0,-6)
```

Create a table for SCAM

```{r}
data.frame(Method="SCAM","Not.Shifted"=sum(as.numeric(f$diffexpressed=="Not Shifted")),"Shifted"=sum(as.numeric(!f$diffexpressed=="Not Shifted")))
```
